### PR TITLE
fix(AWS API Gateway): Ensure that `method` depends on `permission`

### PR DESCRIPTION
**Pull request by nyamba**
Closes: https://github.com/serverless/serverless/issues/9525

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/apiGateway/lib/method/index.js [hunk 1]
```diff
@@ -20,6 +20,9 @@ module.exports = {
         });
       }
 
+      const apiGetwayPermission = this.provider.naming.getLambdaApiGatewayPermissionLogicalId(
```
Comment (pgrzesik) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/method/index.js:23]: there's a small typo - `apiGetway` should be `apiGateway`

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thanks a lot @nyamba, it looks good :+1: It would be, however, great to ensure that this functionality is covered by tests. The tests should land here: https://github.com/serverless/serverless/blob/master/test/unit/lib/plugins/aws/package/compile/events/apiGateway/lib/method/index.test.js and should use `runServerless`

**Comment by nyamba** [comment]
Thanks @pgrzesik, pushed the improvements ;-)

File: test/unit/lib/plugins/aws/package/compile/events/apiGateway/lib/method/index.test.js [hunk 2]
```diff
@@ -1739,6 +1739,36 @@ describe('#compileMethods()', () => {
         .ApiGatewayMethodUsersCreatePost.Properties
     ).to.not.have.key('OperationName');
   });
+
+  it('should depends on permission resource', async () => {
```
Comment (pgrzesik) [inline test/unit/lib/plugins/aws/package/compile/events/apiGateway/lib/method/index.test.js:1743]: Let's move it to this `describe`:
```
describe('#compileMethods v2()', () => {
```

as it's indended to capture all new tests written with `runServerless` approach

File: test/unit/lib/plugins/aws/package/compile/events/apiGateway/lib/method/index.test.js [hunk 3]
```diff
@@ -1739,6 +1739,36 @@ describe('#compileMethods()', () => {
...
+    const {
+      awsNaming,
+      cfTemplate: { Resources: cfResources },
+    } = await runServerless({
+      command: 'package',
+      fixture: 'apiGateway',
+      configExt: {
```
Comment (pgrzesik) [inline test/unit/lib/plugins/aws/package/compile/events/apiGateway/lib/method/index.test.js:1750]: Do we need that configuration extension? The `apiGateway` fixture already configures some events that I believe we can use in this test

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thanks @nyamba - looks good overall, I only have small suggestions, please let me know what do you think :bow:

**Comment by nyamba** [comment]
please check, @pgrzesik

**Review by pgrzesik — APPROVED** [review]
Great changes, thank you @nyamba :bow:

**Pull request merged**